src/sentry/integrations/vsts/issues.py

```diff
@@ -156,3 +157,28 @@ def sync_status_outbound(self, external_issue, is_resolved, project_id, **kwargs
                     'exception': error,
                 }
             )
+
+    def should_unresolve(self, data):
+        return self.is_done(data['old_category']) and not self.is_done(data['new_category'])
+
+    def should_resolve(self, data):
+        return not self.is_done(data['old_category']) and self.is_done(data['new_category'])
+
+    def get_state_categories(self, old_state, new_state, project):
+        client = self.get_client()
+        all_states = client.get_work_item_states(
+            self.instance, project)['value']
+        new_category = None
+        old_category = None
+        for state in all_states:
+            if state['name'] == new_state:
+                new_category = state['category']
+            if state['name'] == old_state:
+                old_category = state['category']
+
+        # what should I do if this doesn't work?
+        assert old_category is not None and new_category is not None
```

Comment:
This is probably not the desired behavior. @macqueen Do you have suggestions?

Reply:
why wouldn't you just treat old/new category as being not "done" if it's`None`?

Reply:
@macqueen because if it's `None` something went wrong. Either the status is wrong or my code was wrong.  VSTS formats it like:
```
newValue: Active
oldValue: New
```
So there's not a possibility to ever have a `None` category unless something went wrong. 

Right now, though we aren't accounting for other types of workitems. I realize the change won't be very hard, but I wasn't aware that it was a priority. Let me know if so. Otherwise I think the above behavior is wrong, but I think we should at least log it. Not sure it's appropriate to act on invalid data...?

Reply:
ok if it's not a scenario that should happen, i think it's fine to raise. I'm not sure I totally follow your last paragraph though -- when is the behavior wrong?

Reply:
I'm sorry. Let me rephrase. I think an ` assert old_category is not None and new_category is not None` is wrong unless caught and given something appropriate to respond with. I assume we don't want a 500 error resulting here. 

I'm not sure about other webhooks, but VSTS actually will automatically disable a webhook if you have a number of unsuccessful responses. I've run into it several times while testing, so it can't be difficult to trigger. We don't currently handle that situation. 

That's my reasoning for not wanting an assertion error, but let me know if there's a different way I should think about it.

Reply:
hmm i mean it seems like we want it to 500 in the case of "my code was wrong" if you're worried about the webhooks being disabled though, you could just log to sentry and then send a 400 response? up to you

Reply:
@macqueen I'm only worried because we don't have any way to handle this. And then the disabled webhook doesn't send any indication it's been disabled. The owner of the VSTS app just gets an email. So...? I'd prefer a 500, but unless the disabled webhook is handled, if anything goes wrong syncing will just stop happening.